rfc-parser/src/main/java/tech/pantheon/yanginator/parser/services/GrammarKitRFCUtils.java

```diff
@@ -0,0 +1,342 @@
+            }
+            result.add(line);
+        }
+
+        return result;
+    }
+
+    /**
+     * Returns a list of string in which string rules are modified. Each string rule
+     * in provided list contains commentaries enclosed in angle brackets ("<...>").
+     * These commentaries are replaced by actual rules.
+     *
+     * @param lines the list of strings in which string rules are to be replaced.
+     * @return the list of strings with string rules replaced.
+     */
+    public static List<String> rewriteStringRules(final List<String> lines) {
+        final Pattern rulePattern = Pattern.compile(".*(?<statementStart>(<[\\s+\\w].*))$");
+        List<String> result = new ArrayList<>();
+        for (int i = 0; i < lines.size(); i++) {
+            String firstLineOfRule = lines.get(i);
+            Matcher matcher = rulePattern.matcher(firstLineOfRule.trim());
+            if (matcher.matches()) {
+                String originalString = matcher.group("statementStart");
+                String quotes = "\"\\\"\"";
+                String secondLineOfRule = lines.get(i + 1);
+                if (secondLineOfRule.contains("-arg")) {
+                    secondLineOfRule = secondLineOfRule.trim().replace(">", quotes);
+                    String replacement = "";
+                    firstLineOfRule = firstLineOfRule.replace(originalString, replacement.concat(quotes).concat(" " + secondLineOfRule));
+                } else {
+                    firstLineOfRule = firstLineOfRule
+                            .replace("<", "//")
+                            .concat(" " + secondLineOfRule.trim())
+                            .replace(">", "");
+                }
+                i++;
+            }
+            result.add(firstLineOfRule);
+        }
+
+        return result;
+    }
+
+    /**
+     * Replaces each occurrence of a hexadecimal range of values with corresponding characters.
+     *
+     * @param lines the list of strings in which range of hexadecimal values will be replaced.
+     * @return the list of strings in which range of hexadecimal values are replaced by characters.
+     */
+    public static List<String> replaceHexadecimalRange(List<String> lines) {
+        List<String> result = new ArrayList<>();
+        final Pattern hexPatternRange = Pattern.compile("(?<hexadecimalRange>(%x\\d{1,2}\\w-\\d{1,2}\\w))");
+
+        for (String line : lines) {
+            Matcher matcher = hexPatternRange.matcher(line.trim());
+            while (matcher.find()) {
+                String originalHexDef = matcher.group("hexadecimalRange").trim();
+                String originalHexDefWithZero = originalHexDef.trim().replaceAll("%", "0");
+                String hexLowerBoundaryString = originalHexDefWithZero.substring(0, 4);
+                int hexLowerBoundary = Integer.decode(hexLowerBoundaryString);
+                int hexUpperBoundary = Integer.parseInt(originalHexDefWithZero.substring(5).toLowerCase(), 16);
```

Comment:
parseInt() does not care about case, so toLowerCase() should not be needed

Reply:
You are right, I will delete toLowerCase().